### DISCUSSION TITLE
str_contains vs Symfony Polyfill / Php80 str_contains

### DISCUSSION THREAD

**Discussion by knobik** [laravel/framework#33386] [General] [answered]
- Laravel Version: 7.16.1
- PHP Version: 7.4.4
- Database Driver & Version: N/A

### Description:
str_contains helper method signature is diffrent then symfony / polyfill for php8 str_contains. It expects both $haystack and $needle to be string, passing array might couse to unexpected errors especially when a random vendor package requires the polyfill and your application does not.

### Solution:
Match the function signature.

**Comment by GrahamCampbell** [comment, marked as answer]
str_contains was removed from the framework core a while ago. We don't recommend you use those functions anymore, but the package is provided for BC.

**Comment by usamaejaz** [comment]
In one of my projects, I also had to deal with this. I ended up replacing str_contains calls with Str::contains

**Reply by GrahamCampbell** [reply]
You should upgrade to Laravel 6 LTS or 8, or downgrade your other dependencies to not use Symfony 8 polyfills.
